lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -736,6 +736,10 @@ void rewriteSlice(SetVector<Value> &slice, DenseMap<Value, Attribute> &layout,
   for (Value v : slice) {
     if (v.getDefiningOp()) {
       opsToRewrite.insert(v.getDefiningOp());
+      if (auto ifOp = v.getDefiningOp<scf::IfOp>()) {
+        opsToRewrite.insert(ifOp.thenYield().getOperation());
+        opsToRewrite.insert(ifOp.elseYield().getOperation());
+      }
```

Comment:
why do we need this special case? Are the yields in the slice?

Reply:
The slice contains values rather than operations, and since yield doesn't return a value I don't think it's possible to represent the yield itself in the slice. This is also true for the `ForOp` where the yield is handled on `BlockArgument` values.